src/Provider/TwigUsageProvider.php

```diff
@@ -0,0 +1,198 @@
+<?php declare(strict_types = 1);
+
+namespace ShipMonk\PHPStan\DeadCode\Provider;
+
+use Composer\InstalledVersions;
+use PhpParser\Node;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Name;
+use PHPStan\Analyser\Scope;
+use PHPStan\BetterReflection\Reflection\Adapter\ReflectionMethod;
+use PHPStan\Node\InClassNode;
+use PHPStan\Reflection\ExtendedMethodReflection;
+use ShipMonk\PHPStan\DeadCode\Graph\ClassMethodRef;
+use ShipMonk\PHPStan\DeadCode\Graph\ClassMethodUsage;
+use ShipMonk\PHPStan\DeadCode\Graph\UsageOrigin;
+use function count;
+use function explode;
+use function in_array;
+
+class TwigUsageProvider implements MemberUsageProvider
+{
+
+    private bool $enabled;
+
+    public function __construct(?bool $enabled)
+    {
+        $this->enabled = $enabled ?? $this->isTwigInstalled();
+    }
+
+    private function isTwigInstalled(): bool
+    {
+        return InstalledVersions::isInstalled('twig/twig');
+    }
+
+    public function getUsages(Node $node, Scope $scope): array
+    {
+        if (!$this->enabled) {
+            return [];
+        }
+
+        $usages = [];
+
+        if ($node instanceof InClassNode) { // @phpstan-ignore phpstanApi.instanceofAssumption
+            $usages = [
+                ...$usages,
+                ...$this->getMethodUsagesFromReflection($node),
+            ];
+        }
+
+        if ($node instanceof New_) {
+            $usages = [
+                ...$usages,
+                ...$this->getMethodUsageFromNew($node, $scope),
+            ];
+        }
+
+        return $usages;
+    }
+
+    /**
+     * @return list<ClassMethodUsage>
+     */
+    private function getMethodUsageFromNew(New_ $node, Scope $scope): array
+    {
+        if (!$node->class instanceof Name) {
+            return [];
+        }
+
+        $className = $node->class->toString();
+
+        if (!in_array($className, [
+            'Twig\TwigFilter',
+            'Twig\TwigFunction',
+            'Twig\TwigTest',
+        ], true)) {
+            return [];
+        }
+
+        $arg = $node->getArgs()[1] ?? null;
```

Comment:
This does not handle named arguments properly, please use normalizer as seen here: https://github.com/shipmonk-rnd/phpstan-rules/blob/bf6c46320f1c1405fa715fcec462f0a6da417703/src/Rule/ForbidCheckedExceptionInCallableRule.php#L452